### PR TITLE
Pawn valid move new

### PR DESCRIPTION
**Pull request by dannypoit**
New version of the `pawn-valid-move` branch/PR.

### CODE REVIEW

File: app/models/pawn.rb [hunk 1]
```diff
@@ -1,2 +1,55 @@
...
+        end
+      end
+    end
+
+    if possible_moves.include?([destination_x, destination_y])
+      return true
+    end
+
+    false
+  end
```
Comment (mscoutermarsh) [inline app/models/pawn.rb]: Lines 39-43 could be reduced to `possible_moves.include?([destination_x, destination_y])` and will have same result.

File: app/models/pawn.rb [hunk 2]
```diff
@@ -1,2 +1,55 @@
...
+        possible_moves << [position_x.to_i, position_y.to_i + 2]
+      end
+    elsif color == 'black'
+      possible_moves << [position_x.to_i, position_y.to_i - 1]
+      if position_y = 6
+        possible_moves << [position_x.to_i, position_y.to_i - 2]
+      end
+    end
+
+    destination_piece = Piece.find_by(position_x: destination_x, position_y: destination_y)
```
Comment (mscoutermarsh) [inline app/models/pawn.rb]: For this method. No need to worry about if there is a piece in the destination square. We can check for that in another method. (Makes things easier on you :smile:)

Eventually, we'll have another method called `move_to!` which will work something like this.

```Ruby
#pseudo code :)
if valid_move?
  return remove_opposing_piece if opposing_piece_in_destination?
  return error if same_color_in_destination?
  move_current_piece
end
```

Comment (dannypoit) [reply app/models/pawn.rb]: @mscoutermarsh Okay, should I also remove the RSpec tests where there's a piece in the destination?

Comment (mscoutermarsh) [reply app/models/pawn.rb]: Up to you. 😄

**Comment by dannypoit** [comment]
I removed the code that checked for a destination piece from the pawn model and the tests that tested this.

My only concern is that diagonal moves only work if there is a piece in the destination, when the pawn is capturing that piece. Will this be added later?

**Pull request merged**